artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/AMQPBrokerConnection.java

```diff
@@ -0,0 +1,556 @@
+      }
+
+      mirrorControlQueue.setMirrorController(true);
+
+      QueueBinding snfReplicaQueueBinding = (QueueBinding)server.getPostOffice().getBinding(replicaConfig.getSourceMirrorAddress());
+      if (snfReplicaQueueBinding == null) {
+         logger.warn("Queue does not exist even after creation! " + replicaConfig);
+         throw new IllegalAccessException("Cannot start replica");
+      }
+
+      Queue snfQueue = snfReplicaQueueBinding.getQueue();
+
+      if (!snfQueue.getAddress().equals(replicaConfig.getSourceMirrorAddress())) {
+         logger.warn("Queue " + snfQueue + " belong to a different address (" + snfQueue.getAddress() + "), while we expected it to be " + addressInfo.getName());
+         throw new IllegalAccessException("Cannot start replica");
+      }
+
+      AMQPMirrorControllerSource newPartition = new AMQPMirrorControllerSource(snfQueue, server, replicaConfig.isMessageAcknowledgements());
+
+      server.scanAddresses(newPartition);
+
+      MirrorController currentMirrorController = server.getMirrorController();
+
+      if (currentMirrorController == null) {
+         server.installMirrorController(newPartition);
+      } else {
+         // Replace a standard implementation by an aggregated supporting multiple targets
+         if (currentMirrorController instanceof AMQPMirrorControllerSource) {
+            // replacing the simple mirror control for an aggregator
+            AMQPMirrorControllerAggregation remoteAggregation = new AMQPMirrorControllerAggregation();
+            remoteAggregation.addPartition((AMQPMirrorControllerSource) currentMirrorController);
+            currentMirrorController = remoteAggregation;
+            server.installMirrorController(remoteAggregation);
+         }
+         ((AMQPMirrorControllerAggregation) currentMirrorController).addPartition(newPartition);
+      }
+
+      return snfReplicaQueueBinding;
+   }
+
+   private void connectReceiver(ActiveMQProtonRemotingConnection protonRemotingConnection,
+                                Session session,
+                                AMQPSessionContext sessionContext,
+                                Queue queue,
+                                Symbol... capabilities) {
+      if (logger.isDebugEnabled()) {
+         logger.debug("Connecting inbound for " + queue);
+      }
+
+      if (session == null) {
+         logger.debug("session is null");
+         return;
+      }
+
+      protonRemotingConnection.getAmqpConnection().runLater(() -> {
+
+         if (receivers.contains(queue)) {
+            logger.debug("Receiver for queue " + queue + " already exists, just giving up");
+            return;
+         }
+         receivers.add(queue);
+         Receiver receiver = session.receiver(queue.getName().toString() + UUIDGenerator.getInstance().generateStringUUID());
```

Comment:
Adding a ":" between the name and generated bit would aid readability when debugging protocol trace.

Reply:
also, the link is named based on the queue name, but then the address name as the source address....I'd tend to expect the receiver link name to incorporate the source address